### DISCUSSION TITLE
Set up development environment

### DISCUSSION THREAD

**Discussion by tvories** [advplyr/audiobookshelf-app#65] [Q&A] [answered]
I saw you mentioned in a previous comment that you hadn't written up the docs yet for how to contribute or set up the development environment.  I'd be happy to lend some help there.

I've worked with native android applications, but I haven't used capacitor yet.  What is the list of frameworks this project uses for mobile development?

**Comment by advplyr** [comment, marked as answer]
Awesome, this is definitely needed. I think a lot more people would help if there was some setup info.

I think capacitor was a good choice, but there are definitely some drawbacks you will notice if you dive in. I wasn't very familiar with webview before this project. Basically 80% of the mobile app can be exactly the same on ios and android, the things we need to handle natively are the audio player, downloading, and android auto/car play.

The capacitor docs are pretty good. https://capacitorjs.com/docs/android
```bash
# git clone and cd into project dir
npm install
npm run generate  # generate static web app
npx cap sync   # copies web app into native android/ios folders
npx cap open android  # open android studio
```
From there you should be able to sync gradle files and run.

The native android code is still a mess with lots of dependencies that need to be cleaned up. I'm not sure if this helps or if you had already gotten this far or what, just let me know. I appreciate your interest in helping.

**Reply by tvories** [reply]
This is very helpful!  Your simple npm install setup is very helpful.  I saw there was an option to use ionic capacitor so I wanted to make sure I wasn't missing any other frameworks.

So the project uses:

1. Vue
2. Capacitor
3. Some native android/ios.

I'll write up a guide on getting the development environment ready for Windows and Mac
